### PR TITLE
Add more support for Omega_h

### PR DESCRIPTION
**Pull request by bartgol**
This PR adds a bit more support for Omega_h:

- Add more common utilities for albany mesh objects setup
- Minor change to OmegahConnMgr interface
- Add omegah version of constrained opt test (only one, for now)
- Fix issue with coord view in omegah mesh

Still working on a few things.

### CODE REVIEW

**Comment by bartgol** [comment]
@cwsmith @mperego Omegah tests now pass on my machine (that is unit test plus the two Albany tests). Please, take a look whenever you have a minute. Thanks!

File: tests/unit/disc/omegah/OmegahConMgr.cpp [hunk 1]
```diff
@@ -33,33 +34,54 @@
...
-auto createOmegahConnManager(Omega_h::Mesh& mesh, std::string partId, const int partDim) {
-  return Teuchos::rcp(new Albany::OmegahConnManager(mesh, partId, partDim));
+auto createOmegahConnManager(const Teuchos::RCP<Albany::OmegahGenericMesh>& mesh,
+                             const std::string& partId) {
+  return Teuchos::rcp(new Albany::OmegahConnManager(mesh, partId));
 }
 
 /* copied from tests/unit/disc/UnitTest_BlockedDOFManager.cpp */
-template <typename Intrepid2Type>
+template <template <typename,typename,typename> class Intrepid2Type>
```
Comment (cwsmith) [inline tests/unit/disc/omegah/OmegahConMgr.cpp:79]: 👍

**Review by cwsmith — APPROVED** [review]
LGTM 👍

**Comment by bartgol** [comment]
Since Cameron approved, and since then I simply rebased on master to fix conflicts, and fixed a tiny bug that slipped through the cracks, I'm going to merge.

**Pull request merged**